src/gameengine/ecs/System.java

```diff
@@ -0,0 +1,12 @@
+package gameengine.ecs;
+
+import java.util.List;
+
+public interface System {
+
+    Class[] requiredComponents();
+
+    Scheduler schedule();
+
+    void update(List<Entity> entities);
```

Comment:
The elapsed time is missing and I wonder if we need to also pass additional information like the context